eogrow/pipelines/download_batch.py

```diff
@@ -162,12 +165,16 @@ def run_procedure(self) -> tuple[list[str], list[str]]:
             return [], []
 
         LOGGER.info("Monitoring batch job with ID %s", batch_request.request_id)
-        results = monitor_batch_job(
-            batch_request,
-            config=self.sh_config,
-            sleep_time=self.config.monitoring_sleep_time,
-            analysis_sleep_time=self.config.monitoring_analysis_sleep_time,
-        )
+        results = self._monitor_job(batch_request)
+
+        # retry partial
+        for _ in range(self.config.num_retries_on_partial):
+            batch_request = self.batch_client.get_request(batch_request)
+            if batch_request.status != BatchRequestStatus.PARTIAL:
```

Comment:
Should this be `BatchRequestStatus.PARTIAL.value` ? I assume `batch_request.status` is a string.

Reply:
in L153 a direct comparison is also made, and the type hint claims its a `BatchRequestStatus`

Reply:
`status: BatchRequestStatus = field(metadata=enum_config(BatchRequestStatus))` the field uses a special constructor